source/xlsxreader.d

```diff
@@ -222,56 +212,61 @@ struct Iterator(T) {
 		return this.data.front;
 	}
 
-	typeof(this) save() {
+	inout(typeof(this)) save() inout pure nothrow @nogc {
 		return this;
 	}
+
+	// Request random access.
+	inout(T)[] array() inout @safe pure nothrow @nogc {
+		return data;
+	}
 }
 
 ///
 struct RowUntyped {
 	Sheet* sheet;
-	/*const*/ size_t row;
-	size_t start;
-	size_t end;
+	const size_t row;
+	const size_t startColumn;
+	const size_t endColumn;
 	size_t cur;
 
-	this(Sheet* sheet, size_t row, size_t start, size_t end) {
+	this(Sheet* sheet, size_t row, size_t startColumn, size_t endColumn) pure nothrow @nogc @safe {
+		assert(sheet.table.length == sheet.maxPos.row + 1);
 		this.sheet = sheet;
 		this.row = row;
-		this.start = start;
-		this.end = end;
-		this.cur = this.start;
+		this.startColumn = startColumn;
+		this.endColumn = endColumn;
+		this.cur = this.startColumn;
 	}
 
-	@property bool empty() const {
-		return this.cur >= this.end;
+	@property bool empty() const pure nothrow @nogc @safe {
+		return this.cur >= this.endColumn;
 	}
 
-	void popFront() {
+	void popFront() pure nothrow @nogc @safe {
 		++this.cur;
 	}
 
-	typeof(this) save() {
+	inout(typeof(this)) save() inout pure nothrow @nogc @safe {
 		return this;
 	}
 
-	@property Cell front() {
+	@property inout(Cell) front() inout pure nothrow @nogc @safe {
 		return this.sheet.table[this.row][this.cur];
 	}
 }
 
 ///
 struct Row(T) {
 	RowUntyped ru;
-
 	T front;
 
-	this(Sheet* sheet, size_t row, size_t start, size_t end) {
-		this.ru = RowUntyped(sheet, row, start, end);
-		this.read();
+	this(Sheet* sheet, size_t row, size_t startColumn, size_t endColumn) {
+		ru = RowUntyped(sheet, row, startColumn, endColumn);
```

Comment:
please re-add the `this.`

Reply:
Done at https://github.com/symmetryinvestments/xlsxreader/pull/16/commits/f45a69687ff008432d9a8fe36dfc550d19d35f31.